tests/functional/broadcast/00-simple.t

```diff
@@ -25,7 +25,9 @@ run_ok "${TEST_NAME_BASE}-validate" cylc validate "${SUITE_NAME}"
 # Debug mode not needed here (and set-x w/ XTRACEFD broken some bash versions)
 suite_run_ok "${TEST_NAME_BASE}-run" \
     cylc run --no-detach --reference-test "${SUITE_NAME}"
-sort "${SUITE_RUN_DIR}/share/broadcast.log" >'broadcast.log.sorted'
+LOG_FILE="${SUITE_RUN_DIR}/share/broadcast.log"
+while [ ! -f "${LOG_FILE}" ]; do sleep 1; done
```

Comment:
Nice spot, thanks.

This has the potential to hang in the (likely) event that something goes wrong with the suite run. This is a common problem with the functional tests so the `test_header` provides a `poll` function with a timeout on whatever you run to avoid hanging tests.

Reply:
Nice! Let me update this PR and then re-run this test a few more times to make sure the test doesn't fail anymore. Thanks @oliver-sanders !